groclient/lib.py

```diff
@@ -606,70 +606,63 @@ def get_geojson(access_token, api_host, region_id, zoom_level):
 
 
 def get_ancestor(access_token, api_host, entity_type, entity_id, distance=None,
-                   include_details=True):
+                 include_details=True, ancestor_level=None, include_historical=True,):
     url = '/'.join(['https:', '', api_host, 'v2/{}/belongs-to'.format(entity_type)])
     headers = {'authorization': 'Bearer ' + access_token}
     params = {'ids': [entity_id]}
     if distance:
         params['distance'] = distance
+    elif ancestor_level:
+        params['level'] = ancestor_level
     else:
         params['distance'] = -1
 
     resp = get_data(url, headers, params)
     ancestor_entity_ids = resp.json()['data'][str(entity_id)]
 
-    if include_details:
+    # Filter out regions with the 'historical' flag set to true
+    if not include_historical or include_details:
         entity_details = lookup(access_token, api_host, entity_type, ancestor_entity_ids)
-        return [entity_details[str(child_entity_id)] for child_entity_id in ancestor_entity_ids]
+
+        if not include_historical:
+            ancestor_entity_ids = [entity['id'] for entity in entity_details.values()
+                                     if not entity['historical']]
+
+        if include_details:
+            return [entity_details[str(child_entity_id)] for child_entity_id in
+                    ancestor_entity_ids]
 
     return [{'id': ancestor_entity_id} for ancestor_entity_id in ancestor_entity_ids]
 
 
 def get_descendant(access_token, api_host, entity_type, entity_id, distance=None,
-                   include_details=True):
+                   include_details=True, descendant_level=None, include_historical=True):
     url = '/'.join(['https:', '', api_host, 'v2/{}/contains'.format(entity_type)])
     headers = {'authorization': 'Bearer ' + access_token}
     params = {'ids': [entity_id]}
     if distance:
         params['distance'] = distance
-    else:
-        params['distance'] = -1
-
-    resp = get_data(url, headers, params)
-    descendant_entity_ids = resp.json()['data'][str(entity_id)]
-
-    if include_details:
-        entity_details = lookup(access_token, api_host, entity_type, descendant_entity_ids)
-        return [entity_details[str(child_entity_id)] for child_entity_id in descendant_entity_ids]
-
-    return [{'id': descendant_entity_id} for descendant_entity_id in descendant_entity_ids]
-
-
-def get_descendant_regions(access_token, api_host, region_id,
-                           descendant_level=None, include_historical=True, include_details=True):
-    url = '/'.join(['https:', '', api_host, 'v2/regions/contains'])
-    headers = {'authorization': 'Bearer ' + access_token}
-    params = {'ids': [region_id]}
-    if descendant_level:
+    elif descendant_level:
```

Comment:
same here, check for entity regions as well

Reply:
right, thanks!